feature/experimental/gribi/ate_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,565 @@
+	ctx        context.Context
+	client     *fluent.GRIBIClient
+	dut        *ondatra.DUTDevice
+	ate        *ondatra.ATEDevice
+	top        *ondatra.ATETopology
+	electionID gribi.Uint128
+}
+
+// createTrafficFlow generates traffic flow from source network to
+// destination network via srcEndPoint to dstEndPoint and checks for
+// packet loss.
+func createTrafficFlow(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology) *ondatra.Flow {
+	ethHeader := ondatra.NewEthernetHeader()
+	ipv4Header := ondatra.NewIPv4Header()
+	ipv4Header.DstAddressRange().
+		WithMin("198.18.196.1").
+		WithMax("198.18.199.255").
+		WithCount(1020)
+
+	srcEndPoint := top.Interfaces()["src"]
+	dstEndPoint := []ondatra.Endpoint{}
+	for intf, intfData := range top.Interfaces() {
+		if intf != "src" {
+			dstEndPoint = append(dstEndPoint, intfData)
+		}
+	}
+
+	flow := ate.Traffic().NewFlow("Flow").
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint...).
+		WithHeaders(ethHeader, ipv4Header)
+
+	return flow
+
+}
+
+// Function to send traffic
+func sendTraffic(t *testing.T, ate *ondatra.ATEDevice, flow *ondatra.Flow) {
+	t.Logf("Starting traffic")
+	ate.Traffic().Start(t, flow)
+}
+
+// Function to verify traffic
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, flow *ondatra.Flow) {
+	flowPath := gnmi.OC().Flow(flow.Name())
+	if got := gnmi.Get(t, ate, flowPath.LossPct().State()); got > 0 {
+		t.Errorf("LossPct for flow %s got %g, want 0", flow.Name(), got)
+	} else {
+		t.Logf("Traffic flows fine from ATE-port1 to ATE-port2")
+	}
+}
+
+// findSecondaryController finds out primary and secondary controllers
+func findSecondaryController(t *testing.T, dut *ondatra.DUTDevice, controllers []string) (string, string) {
+	var primary, secondary string
+	for _, controller := range controllers {
+		role := gnmi.Get(t, dut, gnmi.OC().Component(controller).RedundantRole().State())
+		t.Logf("Component(controller).RedundantRole().Get(t): %v, Role: %v", controller, role)
+		if role == secondaryController {
+			secondary = controller
+		} else if role == primaryController {
+			primary = controller
+		} else {
+			t.Fatalf("Expected controller %s to be active or standby, got %v", controller, role)
+		}
+	}
+	if secondary == "" || primary == "" {
+		t.Fatalf("Expected non-empty primary and secondary Controller, got primary: %v, secondary: %v", primary, secondary)
+	}
+	t.Logf("Detected primary: %v, secondary: %v", primary, secondary)
+
+	return secondary, primary
+}
+
+// Function to stop traffic
+func stopTraffic(t *testing.T, ate *ondatra.ATEDevice) {
+	t.Logf("Stopping traffic")
+	ate.Traffic().Stop(t)
+}
+
+// switchoverReady is to check if controller is ready for switchover
+func switchoverReady(t *testing.T, dut *ondatra.DUTDevice, controller string) bool {
+	switchoverReady := gnmi.OC().Component(controller).SwitchoverReady()
+	_, ok := gnmi.Watch(t, dut, switchoverReady.State(), 30*time.Minute, func(val *ygnmi.Value[bool]) bool {
```

Comment:
I think this can be a simpler gnmi.Await call?

Reply:
updated